src/util/compareThought.ts

```diff
@@ -68,6 +79,28 @@ const toNumber = (x: number | string): number =>
 /** Returns trure if the given string is an integer or decimal number. Recognizes prefixed number strings like "#1" and "$1" as numbers. */
 const isNumber = (x: number | string): boolean => !isNaN(toNumber(x))
 
+/** Checks if a string matches a date pattern like "M/d", "M-d", "M/d/yyyy", "M-d-yyyy", or written formats.
+ * Accepts 1-2 digits for month and day, optionally followed by year
+ * Also accepts written month names like "March 3, 2020" or "December 3, 2020"
+ * Requires consistent separators (all slashes or all dashes)
+ * Examples: "6/21", "6-21", "12/1", "12-1", "6/21/2025", "6-21-2025", "March 3, 2020", "December 3, 2020".
+ */
+export const isDatePattern = (value: string): boolean => {
+  const trimmed = value.trim()
+
+  // Match numeric patterns with consistent separators
+  // Either all slashes: "6/21" or "6/21/2025"
+  const slashPattern = /^\d{1,2}\/\d{1,2}(\/\d{4})?$/
+  // Or all dashes: "6-21" or "6-21-2025"
+  const dashPattern = /^\d{1,2}-\d{1,2}(-\d{4})?$/
+
+  // Match written month patterns: "Month Day, Year" or "Month Day"
+  const writtenPattern =
+    /^(January|February|March|April|May|June|July|August|September|October|November|December)\s+\d{1,2}(,\s*\d{4})?$/i
```

Comment:
While it's nicely contained, regex instantiation is known to be slow, so you'll want to declare them as constants outside the function.